aptos-move/aptos-e2e-comparison-testing/src/execution.rs

```diff
@@ -0,0 +1,374 @@
+            }
+            // execute and compare
+            self.execute_and_compare(
+                cur_version,
+                &state,
+                &features,
+                &txn_idx,
+                compiled_package_cache,
+                compiled_package_cache_v2,
+            );
+        }
+        Ok(())
+    }
+
+    fn execute_and_compare(
+        &self,
+        cur_version: Version,
+        state: &FakeDataStore,
+        features: &Features,
+        txn_idx: &TxnIndex,
+        compiled_package_cache: &mut HashMap<PackageInfo, CompiledPackage>,
+        compiled_package_cache_v2: &mut HashMap<PackageInfo, CompiledPackage>,
+    ) {
+        let mut res_1_opt = None;
+        let mut res_2_opt = None;
+        if self.execution_mode.is_v1() || self.execution_mode.is_compare() {
+            res_1_opt = self.execute_code(
+                state,
+                features,
+                &txn_idx.package_info,
+                &txn_idx.txn,
+                compiled_package_cache,
+            );
+        }
+        if self.execution_mode.is_v2() || self.execution_mode.is_compare() {
+            res_2_opt = self.execute_code(
+                state,
+                features,
+                &txn_idx.package_info,
+                &txn_idx.txn,
+                compiled_package_cache_v2,
+            );
+        }
+        if self.execution_mode.is_compare() {
+            Self::print_mismatches(cur_version, &res_1_opt.unwrap(), &res_2_opt.unwrap());
+        } else {
+            let res = if let Some(res_1) = res_1_opt {
+                res_1
+            } else {
+                res_2_opt.unwrap()
+            };
+            if let Ok(res_ok) = res {
+                println!(
+                    "version:{}\nwrite set:{:?}\n events:{:?}\n",
+                    cur_version, res_ok.0, res_ok.1
+                );
+            } else {
+                println!(
+                    "execution error {} at version: {}, error",
+                    res.unwrap_err(),
+                    cur_version
+                );
+            }
+        }
+    }
+
+    fn execute_code(
+        &self,
+        state: &FakeDataStore,
+        features: &Features,
+        package_info: &PackageInfo,
+        txn: &Transaction,
+        compiled_package_cache: &mut HashMap<PackageInfo, CompiledPackage>,
+    ) -> Option<Result<(WriteSet, Vec<ContractEvent>), VMStatus>> {
+        let executor = FakeExecutor::no_genesis();
+        let mut executor = executor.set_not_parallel();
+        *executor.data_store_mut() = state.clone();
+        if let Transaction::UserTransaction(signed_trans) = txn {
+            let sender = signed_trans.sender();
+            let payload = signed_trans.payload();
+            if let TransactionPayload::EntryFunction(entry_function) = payload {
+                // always load 0x1 modules
+                load_aptos_packages_to_executor(&mut executor, compiled_package_cache);
```

Comment:
Was wondering if this would cause inefficiency issues in the long run if we want to run a huge number of txns. Loading up 0x1 might be a bit slow and if we do that for every txn that might be very inefficient.

Reply:
I changed the code to directly store binary blobs in the compiled_package_cache so that serialization only happens once when the source code is compiled.

Reply:
I think there are two parts of this problem: one is how the compiler reuse existing compilation result. The other is how the VM will be able to cache the Aptos Framework during execution. The latter one is actually a bit harder to solve but just want to flag out this issue. We can look into this problem together later after I've done my loader refactor